soapui/src/main/java/com/eviware/soapui/impl/wsdl/teststeps/assertions/basic/XBaseContainsAssertion.java

```diff
@@ -0,0 +1,304 @@
+package com.eviware.soapui.impl.wsdl.teststeps.assertions.basic;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.swing.JTextArea;
+
+import org.apache.xmlbeans.XmlObject;
+import org.custommonkey.xmlunit.Diff;
+import org.custommonkey.xmlunit.Difference;
+import org.custommonkey.xmlunit.DifferenceEngine;
+import org.custommonkey.xmlunit.DifferenceListener;
+import org.w3c.dom.Node;
+
+import com.eviware.soapui.config.TestAssertionConfig;
+import com.eviware.soapui.impl.wsdl.teststeps.WsdlMessageAssertion;
+import com.eviware.soapui.impl.wsdl.teststeps.WsdlTestRequestStep;
+import com.eviware.soapui.model.TestModelItem;
+import com.eviware.soapui.model.TestPropertyHolder;
+import com.eviware.soapui.model.iface.MessageExchange;
+import com.eviware.soapui.model.iface.SubmitContext;
+import com.eviware.soapui.model.propertyexpansion.PropertyExpansion;
+import com.eviware.soapui.model.propertyexpansion.PropertyExpansionUtils;
+import com.eviware.soapui.model.support.XPathReference;
+import com.eviware.soapui.model.support.XPathReferenceContainer;
+import com.eviware.soapui.model.support.XPathReferenceImpl;
+import com.eviware.soapui.model.testsuite.Assertable;
+import com.eviware.soapui.model.testsuite.AssertionError;
+import com.eviware.soapui.model.testsuite.AssertionException;
+import com.eviware.soapui.model.testsuite.RequestAssertion;
+import com.eviware.soapui.model.testsuite.ResponseAssertion;
+import com.eviware.soapui.model.testsuite.TestProperty;
+import com.eviware.soapui.support.StringUtils;
+import com.eviware.soapui.support.Tools;
+import com.eviware.soapui.support.components.JXToolBar;
+import com.eviware.soapui.support.types.StringList;
+import com.eviware.soapui.support.xml.XmlObjectConfigurationBuilder;
+import com.eviware.soapui.support.xml.XmlObjectConfigurationReader;
+import com.eviware.soapui.support.xml.XmlUtils;
+
+public abstract class XBaseContainsAssertion extends WsdlMessageAssertion implements RequestAssertion, ResponseAssertion,

```

Comment:
I would choose a different name for this assertion, such as AbstractXmlContainsAssertion. This is more consistent with the naming in the SoapUI codebase, and XBase doesn't really mean anything (nor "X").